integration/pull_test.go

```diff
@@ -796,8 +794,43 @@ min_layer_size=` + minLayerSize + `
 
 	sh.X("soci", "image", "rpull", "--user", regConfig.creds(), "--soci-index-digest", indexDigest, regConfig.mirror(containerImage).ref)
 
-	// default index creation min layer size of 10MB would produce 3 fuse mounts
-	// specified index creation min layer size of 0 would produce 10 fuse mounts
-	// specified runtime min layer size of 25MB will produce 2 fuse mounts
-	checkFuseMounts(t, sh, 2)
+	var snapshotSizes []int64
+	snapshotListOutput := string(sh.O("ctr", "snapshot", "list"))
+	snapshotList := strings.Split(snapshotListOutput, "\n")
+	for _, snapshotLine := range snapshotList[1 : len(snapshotList)-1] {
+		snapshotDigest := strings.Split(snapshotLine, " ")[0]
+		rawJSON := sh.O("ctr", "snapshot", "--snapshotter", "soci", "info", snapshotDigest)
+		var snapshotInfo snapshots.Info
+		if err := json.Unmarshal(rawJSON, &snapshotInfo); err != nil {
+			t.Fatalf("invalid snapshot info from digest %s: %v", snapshotDigest, rawJSON)
+		}
+		strVal := snapshotInfo.Labels[source.TargetSizeLabel]
+		intVal, err := strconv.ParseInt(strVal, 10, 64)
+		if err != nil {
+			t.Fatalf("snapshot size label cannot be converted to int: %s", strVal)
+		}
+		snapshotSizes = append(snapshotSizes, intVal)
+	}
+
+	sort.Slice(snapshotSizes, func(i, j int) bool { return snapshotSizes[i] < snapshotSizes[j] })
+	medianIndex := int(len(snapshotSizes) / 2)
+	medianSize := snapshotSizes[medianIndex]
+	// find the lowest index that is the same size as the median
+	for medianIndex > 0 && snapshotSizes[medianIndex-1] == medianSize {
+		medianIndex--
+	}
+
+	minLayerSizeConfig := `
+[snapshotter]
+min_layer_size=` + strconv.FormatInt(medianSize, 10) + `
+`
+
+	rebootContainerd(t, sh, getContainerdConfigToml(t, false), getSnapshotterConfigToml(t, false, minLayerSizeConfig))
```

Comment:
It may be possible to partially reset the environment here instead, to avoid needing to copy the image and build the index again, just making the environment clean for another rpull. I made a moderate effort in that direction but decided this more concise and certain-to-work solution would be acceptable. Feedback on this decision is welcome.

Reply:
Opened #401 based on this and other resolved comment below.